### DISCUSSION TITLE
추상 클래스와 인터페이스의 차이

### DISCUSSION THREAD

**Discussion by uneap** [11st-corp/book-club#21] [Q&A]
두가지의 차이

**Comment by uneap** [comment]
모든 메서드가 추상 메소드로 이루어진 것이 인터페이스
추상클래스는 일반 클래스에 추상 메소드가 포함되어있는 것.
추상클래스의 목적: 상속을 받아서 기능을 확장 시키는 것
인터페이스의 목적: 구현하는 모든 클래스에 대해 특정한 메서드가 반드시 존재하도록 강제. 구현 객체의 같은 동작을 보장하기 위한 목적 예를 들면 차 interface일 경우 차의 기능을 가지고 있는 것을 보장한다는 것? 그룹화 하는 목적이라는 것인가?

**Reply by KoEonYack** [reply]
https://github.com/brave-people/brave-tech-interview/blob/9ef86cf2acb228185a5542139d60bd3e6d034593/contents/language.md#interface%EC%99%80-abstract%EC%97%90-%EB%8C%80%ED%95%B4%EC%84%9C-%EC%84%A4%EB%AA%85%ED%95%98%EC%84%B8%EC%9A%94---1%ED%8E%B8
